src/Execution/MutationExecutor.php

```diff
@@ -156,9 +201,35 @@ public static function executeUpdate(Model $model, Collection $args, HasMany $pa
      */
     protected static function extractBelongsToArgs(Model $model, Collection $args): Collection
     {
-        return $args->partition(function ($value, string $key) use ($model) {
-            return method_exists($model, $key)
-                && ($model->{$key}() instanceof BelongsTo);
+        return $args->partition(function ($value, $key) use ($model) {
+            return method_exists($model, $key) && ($model->{$key}() instanceof BelongsTo);
+        });
+    }
+
+    /**
+     * @param Model $model
+     * @param Collection $args
+     *
+     * @return Collection
+     */
+    protected static function extractMorphToArgs(Model $model, Collection $args): Collection
+    {
+        return $args->partition(function ($value, $key) use ($model) {
```

Comment:
Consider extracting the partitioning into its own method and pass the type of relation as an argument.